Machine_Learning/notebooks/czmodel/Regresssion_PyTorch_5_0_0.ipynb

```diff
@@ -0,0 +1,676 @@
+    "\n",
+    "    def __getitem__(self, idx):\n",
+    "        image = self.sample_images[idx]\n",
+    "        label = self.sample_labels[idx]\n",
+    "        img = cv2.imread(image, cv2.IMREAD_GRAYSCALE)\n",
+    "        labl = cv2.imread(label, cv2.IMREAD_GRAYSCALE) \n",
+    "\n",
+    "        if self.transforms is not None:\n",
+    "            img = self.transforms(img)\n",
+    "            labl = self.transforms(labl)\n",
+    "\n",
+    "        return img.to(torch.float32), labl.to(torch.float32)\n",
+    "\n",
+    "    def __len__(self):\n",
+    "        return len(self.sample_images)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Define transforms\n",
+    "from torchvision import transforms\n",
+    "transforms = transforms.Compose([\n",
+    "    transforms.ToTensor(),\n",
+    "    transforms.CenterCrop(CROP_SIZE),\n",
+    "])\n",
+    "\n",
+    "# Create dataset and dataloader\n",
+    "sample_dataset = SampleDataset(images_folder=IMAGES_FOLDER,\n",
+    "                               labels_folder=LABELS_FOLDER,\n",
+    "                               transforms=transforms)\n",
+    "\n",
+    "sample_dataloader = DataLoader(dataset=sample_dataset,\n",
+    "                                      batch_size=2,\n",
+    "                                      shuffle=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "ZRujNv7nrEEG"
+   },
+   "source": [
+    "### Define a simple model\n",
+    "This part defines a simple PyTorch convolutional model for regression task."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "prtTh7t2ZubC"
+   },
+   "outputs": [],
+   "source": [
+    "# Define simple PyTorch convolutional model\n",
+    "class SampleModel(nn.Module):\n",
+    "\n",
+    "    def __init__(self):\n",
+    "        super(SampleModel, self).__init__()\n",
+    "        self.conv = nn.Sequential(\n",
+    "            # Encoder\n",
+    "            nn.Conv2d(1, 16, 3, padding=1),  \n",
+    "            nn.ReLU(),\n",
+    "            nn.MaxPool2d(2, padding=0),\n",
+    "            nn.Conv2d(16, 16, 3, padding=1),  \n",
+    "            nn.ReLU(),\n",
+    "            nn.MaxPool2d(2, padding=0),\n",
+    "            nn.Conv2d(16, 16, 3, padding=1),  \n",
+    "            nn.ReLU(),\n",
+    "            # Decoder\n",
+    "            nn.Upsample(scale_factor=2, mode='bilinear'),\n",
+    "            nn.Conv2d(16, 16, 3, padding=1),  \n",
+    "            nn.ReLU(),\n",
+    "            nn.Upsample(scale_factor=2, mode='bilinear'),\n",
+    "            nn.Conv2d(16, 1, 3, padding=1),              \n",
+    "        )\n",
+    "\n",
+    "    def forward(self, x):\n",
+    "        x = self.conv(x)\n",
+    "        return x\n",
+    "\n",
+    "\n",
+    "model = SampleModel().to(device)"
```

Comment:
![image](https://user-images.githubusercontent.com/9250590/190161017-61a8f0fd-284b-4d32-9650-f01a0217db9a.png)

Reply:
Could you check if this cell is run and what is the output? 
![image](https://user-images.githubusercontent.com/85402831/190161622-f8aeac33-d449-4216-92da-ff538b27d331.png)

Reply:
My fault.